src/config/config_render.go

```diff
@@ -0,0 +1,398 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/shibukawa/configdir"
+	"github.com/simelo/rextporter/src/common"
+	"github.com/spf13/viper"
+)
+
+type templateData struct {
+	ServiceConfigPath string
+	MetricsConfigPath string
+}
+
+type mainConfigData struct {
+	mainConfigPath string
+	tmplData       templateData
+}
+
+func (confData mainConfigData) ServiceConfigPath() string {
+	return confData.tmplData.ServiceConfigPath
+}
+
+func (confData mainConfigData) MetricsConfigPath() string {
+	return confData.tmplData.MetricsConfigPath
+}
+
+func (confData mainConfigData) MainConfigPath() string {
+	return confData.mainConfigPath
+}
+
+const mainConfigFileContentTemplate = `
+serviceConfigTransport = "file" # "file" | "consulCatalog"
+# render a template with a portable path
+serviceConfigPath = "{{.ServiceConfigPath}}"
+metricsConfigPath = "{{.MetricsConfigPath}}"
+`
+
+const serviceConfigFileContentTemplate = `
+# Service configuration.
+name = "wallet"
+scheme = "http"
+port = 8000
+basePath = ""
+authType = "CSRF"
+tokenHeaderKey = "X-CSRF-Token"
+genTokenEndpoint = "/api/v1/csrf.json"
+tokenKeyFromEndpoint = "csrf_token"
+
+[location]
+  location = "localhost"
+`
+const metricsConfigFileContentTemplate = `
+# All metrics to be measured.
+[[metrics]]
+  name = "seq"
+  url = "/api/v1/health.json"
+  httpMethod = "GET"
+  path = "/blockchain/head/seq"
+
+  [metrics.options]
+    type = "Counter"
+    description = "I am running since"
+
+# [[metrics]]
+#   name = "openConnections"
+#   url = "/api/v1/network/connections"
+#   httpMethod = "GET"
+#   path = "/"
+
+#   [metrics.options]
+#     type = "Histogram"
+#     description = "Connections ammount"
+
+#   [metrics.histogramOptions]
+#     buckets = [
+#       1,
+#       2, 
+#       3
+#     ]
+
+
+
+
+# TODO(denisacostaq@gmail.com):
```

Comment:
@denisacostaq services should be match against whole templates , rather than individual metrics . The match is not for hosts directly . The match is for service types . Works like this 

`metrics <- m:1 -> templates <- m:n -> services <- 1:n -> (physical | virtual) hosts`